### PR TITLE
Very rough+small start, demoed a few weeks ago with VoiceOver.

### PR DESCRIPTION
**Pull request by pbarry-r7**
EDIT: this is in reference to this [Metasploit Framework discussion](https://github.com/rapid7/metasploit-framework/discussions/14367#discussioncomment-592279).

This was a hackathon project I did in early 2020 (also explains why this fork+branch is WAY behind current Framework) where I used the built-in macOS screen reader.  With limited time (hackathon!), I  looked at improving three problems I noticed right away:

1. Reading useless stuff
2. Reading without appropriate pause
3. No indication of missing data

I put my work behind a new `msfconsole` commandline option: `-s`.  Then I made some improvements (hopefully), such as adding commas/punctuation to create pauses and `NIL` to convey that a field was empty.  Then I juxstaposed some common commands with "before" and "after" video clips, where "before" would be how VoiceOver currently (or in early 2020, anyway) would read `msfconsole` output, and "after" would be running the same command but in a session of `msfconsole -s` where I had tried to make it better.  Here's some of those clips (MAKE SURE TO UNMUTE EACH CLIP!  :P  ):

### starting the console up (wanted to still read the data around module counts, but skip the ASCII banner stuff)

BEFORE

https://user-images.githubusercontent.com/19912177/114250968-0c2db200-9965-11eb-9f66-bae7b6b66e45.mp4

AFTER

https://user-images.githubusercontent.com/19912177/114250973-12bc2980-9965-11eb-8b4a-836f872cd5c1.mp4

### `hosts` command 

BEFORE

https://user-images.githubusercontent.com/19912177/114251051-54e56b00-9965-11eb-9151-b5fdd9f2fb60.mp4

AFTER

https://user-images.githubusercontent.com/19912177/114251202-cf15ef80-9965-11eb-8e1e-14a603f759d6.mp4

### listing a handler's advanced options

BEFORE

https://user-images.githubusercontent.com/19912177/114251125-878f6380-9965-11eb-9089-3c5b9b04d853.mp4

AFTER

https://user-images.githubusercontent.com/19912177/114251123-83634600-9965-11eb-9da4-3deaf85c128e.mp4

### CODE REVIEW

**Comment by adfoster-r7** [comment]
Looks/sounds awesome. I wonder if we could make the screen reader changes directly to rex-text's table generation directly instead of modifying all of the tables directly in framework directly - that might reduce the effort of getting an MVP released 🤔

File: lib/msf/base/serializer/readable_text.rb [hunk 1]
```diff
@@ -591,16 +591,21 @@ def self.dump_advanced_options(mod, indent = '')
...
       'Columns' =>
         [
-          'Name',
-          'Current Setting',
-          'Required',
-          'Description'
+          'Name,',
+          'Current Setting,',
+          'Required,',
+          'Description,.'
```
Comment (jmartin-r7) [inline lib/msf/base/serializer/readable_text.rb:597]: A suggestion, table additions for punctuation may be reasonable to implement as an option passed to the `Rex::Text::Table` base on a if `screen reading` is enabled.

Comment (pbarry-r7) [reply lib/msf/base/serializer/readable_text.rb:597]: Indeed, exactly where I was thinking made the most sense to put this capability.

**Comment by bcoles** [comment]
Replacing `print_` method progress indicators with words may also be a good idea.

I get mixed results running these through TTS programs:

```
[+] Something good happened
[-] Something bad happened
[*] Something notable happened
```

**Pull request closed without merging**